internal/audit/cluster_checks.go

```diff
@@ -214,3 +218,57 @@ func makeCheckClusterHighHAAssets(haAssetsThreshold int) checkFunc {
 		return Pass, nil
 	}
 }
+
+func checkClusterNamesForWhitespace(reader *archive.Reader, examples *ExamplesCollection) (Outcome, error) {
+
+	for _, clusterName := range reader.GetClusterNames() {
+		if strings.ContainsFunc(clusterName, unicode.IsSpace) {
```

Comment:
It's worth stating again the problem we're facing is just actual space, not things that look like space but actual space.  Space has significance in protocol parsing and so the message is parsed incorrectly.  Things that look like space is not significant.

In 2.11 we will probably add constraints to make these names be `restricted-term` as defined by ADR-6

Reply:
Ah, I see. Just spaces? What about newlines, for example?

Reply:
I doubt it would be possible to enter a newline config parser would probably fail or chop it

Reply:
so bad, ok we should probably check for that ugh - maybe we should just check for anything thats not `restricted-term`?

Reply:
cc @wallyqs just in case there's some more restrictions we should put in as newline for sure would cause protocol issues

Reply:
@ripienaar, are there any "whitespace" characters filtered out by `unicode.IsSpace` that could/should be used in a server name?

```
// IsSpace reports whether the rune is a space character as defined
// by Unicode's White Space property; in the Latin-1 space
// this is
//
//	'\t', '\n', '\v', '\f', '\r', ' ', U+0085 (NEL), U+00A0 (NBSP).
//
// Other definitions of spacing characters are set by category
// Z and property Pattern_White_Space.
func IsSpace(r rune) bool {
```

Reply:
Probsbly not but the reason I am trying to be minimalist here is that our initial aim is to not break LTS users in cases we do not need to 

this means we want to allow potentially bad ideas as long as they don’t break NGS when they connect to it. 

Later - 2.11 timeline and after discussions - we will decide what the new acceptable names are. 

So yea checking for extra things is not a problem and not bad - but our current aim is to capture the bare minimum cases that cause problems so we break as few as possible users.

Reply:
Well this check in `audit` doesn't break anything. 

It may be worth being a bit more pedantic ahead of the actual changes to server.